tests/integration/feeds/test_feeds_client.py

```diff
@@ -0,0 +1,249 @@
+"""Integration tests for FeedsClient."""
+
+from pathlib import Path
+from random import randint
+from typing import BinaryIO, Callable
+
+import pytest
+from nisystemlink.clients.core import ApiException
+from nisystemlink.clients.feeds import FeedsClient
+from nisystemlink.clients.feeds.models import CreateFeedRequest, Platform
+
+
+FEED_DESCRIPTION = "Sample feed for uploading packages"
+PACKAGE_PATH = (
+    Path(__file__).parent.resolve()
+    / "test_files"
+    / "sample-measurement_0.5.0_windows_x64.nipkg"
+)
+INVALID_PACKAGE_PATH = Path(__file__).parent.resolve()
+
+
+@pytest.fixture(scope="class")
+def client(enterprise_config) -> FeedsClient:
+    """Fixture to create a FeedsClient instance."""
+    return FeedsClient(enterprise_config)
+
+
+@pytest.fixture(scope="class")
+def create_feed(client: FeedsClient):
+    """Fixture to return a object that creates feed."""
+    feed_ids = []
+
+    def _create_feed(feed):
+        response = client.create_feed(feed)
+        feed_ids.append(response.id)
+        return response
+
+    yield _create_feed
+
+    # deleting the created feeds.
+    for feed_id in feed_ids:
+        client.delete_feed(id=feed_id)
+
+
+@pytest.fixture(scope="class")
+def create_feed_request():
+    """Fixture to create a request body of create feed API."""
+
+    def _create_feed_request(feed_name: str, description: str, platform: Platform):
+        feed_request = CreateFeedRequest(
+            name=feed_name,
+            platform=platform,
+            description=description,
+        )
+        return feed_request
+
+    yield _create_feed_request
+
+
+@pytest.fixture(scope="class")
+def binary_pkg_file_data() -> BinaryIO:
+    """Fixture to return package file in binary format."""
+    package_data = open(PACKAGE_PATH, "rb")
+    return package_data
+
+
+@pytest.fixture(scope="class")
+def invalid_id() -> str:
+    """Generate a invalid id."""
+    id = f"Invalid-id-{randint(1000, 9999)}"
+    return id
+
+
+@pytest.fixture(scope="class")
+def get_feed_name():
+    """Generate a feed name."""
+    name = "Test_Feeds_"
+    feed_count = 0
+
+    def _get_feed_name():
+        nonlocal feed_count
+        feed_count += 1
+        feed_name = name + str(feed_count)
+        return feed_name
+
+    yield _get_feed_name
+
+
+@pytest.mark.enterprise
+@pytest.mark.integration
+class TestFeedsClient:
+    """A set of test methods to test SystemLink Feeds API."""
+
+    def test__create_feed_windows_platform__succeeds(
+        self,
+        create_feed: Callable,
+        create_feed_request: Callable,
+        get_feed_name: Callable,
+    ):
+        """Test the case of a completely successful create feed API for Windows platform."""
+        request_body = create_feed_request(
+            feed_name=get_feed_name(),
+            description=FEED_DESCRIPTION,
+            platform=Platform.WINDOWS,
+        )
+        response = create_feed(request_body)
```

Comment:
Its cleaned because of the `create_feed` py fixture which stores the list of feed_ids created and deletes all the feed ids, during the closure of the class instance.

Is it fine then @santhoshramaraj ?

Reply:
Ah, okay, I overlooked and have been focussing on the `create_feed_request()` fixture alone.